api/api-dashboard-v2/src/main/java/com/thoughtworks/go/apiv2/dashboard/representers/PipelineRepresenter.java

```diff
@@ -33,25 +35,25 @@
 
     public static void toJSON(OutputWriter jsonOutputWriter, GoDashboardPipeline model, Username username) {
         String usernameString = username.getUsername().toString();
-
         jsonOutputWriter
-            .addLinks(linksWriter -> addLinks(linksWriter, model))
-            .add("name", model.name().toString())
-            .add("last_updated_timestamp", model.getLastUpdatedTimeStamp())
-            .add("locked", model.model().getLatestPipelineInstance().isCurrentlyLocked())
-            .addChild("pause_info", getPauseInfoNEW(model))
-            .add("can_operate", model.isPipelineOperator(usernameString))
-            .add("can_administer", model.canBeAdministeredBy(usernameString))
-            .add("can_unlock", model.canBeOperatedBy(usernameString))
-            .add("can_pause", model.canBeOperatedBy(usernameString));
+                .addLinks(linksWriter -> addLinks(linksWriter, model))
+                .add("name", model.name().toString())
+                .add("last_updated_timestamp", model.getLastUpdatedTimeStamp())
+                .add("locked", model.model().getLatestPipelineInstance().isCurrentlyLocked())
+                .addChild("pause_info", getPauseInfoNEW(model))
+                .add("can_operate", model.isPipelineOperator(usernameString))
+                .add("can_administer", model.canBeAdministeredBy(usernameString))
+                .add("can_unlock", model.canBeOperatedBy(usernameString))
+                .add("can_pause", model.canBeOperatedBy(usernameString))
+                .addChild("origin", getOrigin(model.getOrigin()));
```

Comment:
If all we're doing is to disable the edit button, can I suggest that we use the `can_administer` or just make this field a boolean to reduce the size and cost of serialization?

Reply:
@ketan -- can we handle only the permissions related stuff with `can_administer`?
Regarding sending the origin attribute, some more context is provided on issue #4403.

We will require the origin attribute to show more information about the config repo in which the pipeline is defined.

Reply:
Since we don't know the specifics that are needed, can we leave it be for now?